src/main/protobuf/game.proto

```diff
@@ -2,19 +2,237 @@ syntax = "proto3";
+  // Should we add another rpc call for acknowledge other than TakeAction?
+  ACKNOWLEDGE = 19; 
+}
+
+/**
+ * Encapsulates a GameEvent response from the Catan game server.
+ * All fields are optional (leaving the action blank will default it to the enum value 0 GAME_ACTION_NONE).
+ * Depending on the type of action, different combinations of fields will be specified.
+ * Examples:
+ * GameEvent {
+ *   position:-1, 
+ *   action: GAME_ACTION_NONE,
+ *   specification: null,
+ *   result: null,
+ *   message: "GAME END"
+ *  }
+ * GameEvent {
+ *   position: 2
+ *   action: MOVE_ROBBER_AND_STEAL,
+ *   specification: {hex: B4, other_player_position: 1},
+ *   result: {card: HiddenCard{ viewable_by_positions: [1, 2], encrypted_name: "askdgi4858fjmcd39932"}},
+ *   message: null
+ * }
+ * GameEvent {
+ *   position: 2
+ *   action: ACTIVATE_MONOPOLY,
+ *   specification: {ask: ORE},
+ *   result: {resources_transacted: {
+ *     1: {gain: [], lose: [ORE, ORE, ORE, ORE]},
+ *     2: {gain: [ORE, ORE, ORE, ORE, ORE, ORE], lose: []},
+ *     3: {gain: [], lose: [ORE, ORE]},
+ *     4: {gain: [] lose: []},
+ *   message: null
+ * }
+ * GameEvent {
+ *   position: 3, 
+ *   action: BUILD_CITY,
+ *   specification: {vertex: B4R},
+ *   result: {resources_transacted: {
+ *     1: {gain: [], lose: []},
+ *     2: {gain: [], lose: []},
+ *     3: { gain: [], lose: [ORE, ORE, ORE, WHEAT, WHEAT]},
+ *     4: {gain: [] lose: []},
+ *   message: null
+ *  }
+ * Important to note, is that multiple actions are able to recycle specification.
+ * For example, BUILD_CITY and BUILD_ROAD have the same specification.
+ * Similarly proposing a trade, making a trade, playing a monopoly or a year of plenty
+ * all take advantage of one or both of the `ask` and `give` fields in the ActionSpecification.
+ **/
+message GameEvent {
+  int32 position = 1; // Player positions on the server need to be 1 not 0 based, so 0 can indicate no position, or we can pass -1
+  GameAction action = 2;
+  ActionSpecification specification = 3;
+  ActionResult result = 4;
+  string message = 5;
+}
+
+/**
+ * Defines the action a player wants to make.
+ * Is used both in the TakeAction RPC and in the GameEvent response.
+ * Of note, like the GameAction enum, the ActionSpecification object used in the TaekAction RPC
+ * is directly piped through into the GameEvent response.
+ * The GameEvent response might optionally also inlcude a result.
+ * For more inforamtion see GameEvent.
+ * Depending on the action being taken, different combinations of the below fields will be specified.
+ * See mapping below:
+ * GAME_ACTION_NONE --> No Action Spec
+ * INITIAL_PLACEMENT --> vertex, edge
+ * ROLL_DICE --> No Action Spec
+ * MOVE_ROBBER_AND_STEAL --> hex, other_player
+ * BUILD_ROAD --> edge
+ * BUILD_SETTLEMENT --> vertex
+ * BUILD_CITY --> vertex
+ * BUILD_DEVELOPMENT_CARD --> No Action Spec
+ * ACTIVATE_KNIGHT --> hex, other_player
+ * ACTIVATE_ROAD_BUILDING --> edges
```

Comment:
Road Builder might be used to only build one road if the player only has one road left. If they dont have any roads left I think they cant play road builder

Reply:
Good call out. I'll include that in the design doc / table I'm making.